controllers/grid/plugins/form/UploadPluginForm.inc.php

```diff
@@ -83,9 +85,10 @@ function execute($request) {
 						__('manager.plugins.installSuccessful', array('versionNumber' => $pluginVersion->getVersionString(false))))
 				);
 			} else if ($this->_function == PLUGIN_ACTION_UPGRADE) {
+				$plugin = PluginRegistry::getPlugin($request->getUserVar('category'), $request->getUserVar('plugin'));
```

Comment:
Just checking to make sure that there aren't any assumptions made about the validity of the input data. I just looked, and it's fine -- it just tries to fetch a plugin object from an array using the specified parameters as indexes.